src/lib/videotagging_extensions/detection.js

```diff
@@ -0,0 +1,193 @@
+            self.detectionAlgorithmManager.setReviewer(method, modelPath);
+            self.detectionAlgorithmManager.reviewer.reviewImagesFolder(reviewPath).then( (modelTags) => {
+                self.videotagging.frames = [];
+                self.videotagging.optionalTags.createTagControls(Object.keys(modelTags.classes));
+
+                //Create regions based on the provided modelTags
+                Object.keys(modelTags.frames).map( (pathId) => {
+                    var frameImage = new Image();
+                    frameImage.src = `${reviewPath}\\${pathId}`;
+                    frameImage.onload = loadFrameRegions; 
+
+                    function loadFrameRegions() {
+                        var imageWidth = this.width;
+                        var imageHeight = this.height;
+                        frameId = pathId.replace(".jpg", "");//remove.jpg
+                        self.videotagging.frames[frameId] = [];
+                        modelTags.frames[pathId].regions.forEach( (region) => {
+                            self.videotagging.frames[frameId].push({
+                            x1:region.x1,
+                            y1:region.y1,
+                            x2:region.x2,
+                            y2:region.y2,                          
+                            id:self.videotagging.uniqueTagId++,
+                            width:imageWidth,
+                            height:imageHeight,
+                            type:self.videotagging.regiontype,
+                            tags:Object.keys(modelTags.classes).filter( (key) => {return modelTags.classes[key] === region.class }),
+                            name:(self.videotagging.frames[frameId].length + 1),
+                            blockSuggest: true
+                            }); 
+                        });
+                    }
+                });
+                self.videotagging.showAllRegions();
+                //cleanup and notify
+                self.videotagging.video.currentTime = 0;
+                self.videotagging.playingCallback();
+                let notification = new Notification('Offline Video Tagger', { body: 'Model Ready For Review.' });
+                cb();
+            });
+        }
+
+        function saveFrame(frameId, fCanvas, canvasContext, saveCb){
+            canvasContext.drawImage(videotagging.video, 0, 0);
+            var writePath =  `${reviewPath}/${frameId}.jpg`
+            var data = fCanvas.toDataURL('image/jpeg').replace(/^data:image\/\w+;base64,/, ""); // strip off the data: url prefix to get just the base64-encoded bytes http://stackoverflow.com/questions/5867534/how-to-save-canvas-data-to-file
+            var buf = new Buffer(data, 'base64');
+            //write canvas to file and change frame
+            console.log('saving file', writePath);
+            if (!fs.existsSync(writePath)) {
+                fs.writeFileSync(writePath, buf);
```

Comment:
Does it have to be Sync? It's not the end of the world but since you already have a callback here that you call it might be better you use the async version and call the saveCb when it finished.